### PR TITLE
Niroy githp

### PR DESCRIPTION
**Pull request by niroyb**
Add instructions to install git on HP (v3) systems.

### CODE REVIEW

**Comment by niroyb** [comment]
@MSFTOSSMgmt/ostc-devs

**Comment by ericgable** [comment]
Looks Good!  Signed off.

**Comment by jeffaco** [comment]
I have some comments, stand by.

File: gitinstall-hp.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Installing git on HP-UX
+
+## Using Depot packages
```
Comment (jeffaco) [inline gitinstall-hp.md]: Please add a table of contents section. See other pages for installing git to see how I did this.

File: gitinstall-hp.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+* Create a temporary folder to hold the depot files : `mkdir -p /root/GIT`
+* Copy over the depot packages. [HPv3 ia64](\\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3ia64), [HPv3 parisc](\\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3risc)
```
Comment (jeffaco) [inline gitinstall-hp.md]: This link doesn't appear to work. As I recall, I had problems with this too. If you can't get it to work, then put it in the document with a `\path\like\this`.

File: gitinstall-hp.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+swinstall -s /root/GIT/openldap-2.4.43-ia64-11.31.depot openldap
+swinstall -s /root/GIT/openssl-1.0.2e-ia64-11.31.depot openssl
+swinstall -s /root/GIT/packages-ia64-11.31 packages
+swinstall -s /root/GIT/perl-5.10.1-ia64-11.31.depot perl
+swinstall -s /root/GIT/tcltk-8.5.18-ia64-11.31.depot tcltk
+swinstall -s /root/GIT/termcap-1.3.1-ia64-11.31.depot termcap
+swinstall -s /root/GIT/zlib-1.2.8-ia64-11.31.depot zlib
+
+swinstall -s /root/GIT/git-2.7.0-ia64-11.31.depot git
+```
```
Comment (jeffaco) [inline gitinstall-hp.md:53]: Do these files need to be updated? If so, please work with me on that. This should be part of the work to commit this (i.e. don't commit until the share is up to date).

File: gitinstall-hp.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+* Test the installation by trying to clone a repo, create a commit in a new branch and push it upstream.
+* Cleanup : `rm -rf /root/GIT`
+
+## Using depothelper
```
Comment (jeffaco) [inline gitinstall-hp.md]: So I also want to know how to construct the files if, for some reason, we need to update. See what I did for Solaris. I have files that can be downloaded, and I explained (with links) how I generated these files if you need to update them.

File: gitinstall-hp.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+* Get the correct depothelper version for your flavor of HP from [hpux.connect.org](http://hpux.connect.org.uk/hppd/hpux/Sysadmin/depothelper-2.00/)
+* Install depothelper for example: `swinstall -s depothelper-2.00-hppa-11.31.depot depothelper`
+* Install git with depothelper : `/usr/local/bin/depothelper git`
+* Test the installation by trying to clone a repo, create a commit in a new branch and push it upstream.
```
Comment (jeffaco) [inline gitinstall-hp.md]: Is depothelper the suggested method of installing? Or installing from the repo files stored on the wss share? Please make it clear what is suggested, but make it clear when/why you'd want to move to other means.

File: gitinstall-hp.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
+# Installing git on HP-UX
+
+## Using Depot packages
+
+* Create a temporary folder to hold the depot files : `mkdir -p /root/GIT`
+* Copy over the depot packages. [HPv3 ia64](\\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3ia64), [HPv3 parisc](\\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3risc)
+* Unzip the compressed depot files : `gunzip /root/GIT/*.gz`
+* Install the git dependencies then git itself. The dependencies were obtained using depothelper.
```
Comment (jeffaco) [inline gitinstall-hp.md]: How? Is there a documented method you can refer to like I did with Solaris?

File: gitinstall-hp.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+It is possible that the git binary installed through depothelper does not work because the configuration of your system is different. Simply rebuild git from source.
+* Get the appropriate HP ported version of git source code from [hpux.connect.org](http://hpux.connect.org.uk/hppd/hpux/Development/Tools/git-2.7.0/). It is the one that has the package type : Source Code.
+* Although the file is named "tar.gz" it might just be a regular tar file. Verify with `file git-2.7.0-src-11.31.tar.gz`
+* Untar the git source `tar -x git-2.7.0-src-11.31.tar.gz`
+* Configure git for your system : `cd git-2.7.0/; ./configure`
+* Build git : `make -j`
+* Make sure an old version of git is not present : `sudo swremove git`
+* Install git : `sudo make install`
+* Test the installation by trying to clone a repo, create a commit in a new branch and push it upstream.
+* If the basic functionality test fails, you might have to reconfigure git using non default options.
```
Comment (jeffaco) [inline gitinstall-hp.md]: Do you have "basic functionality tests" that you used? Perhaps you should quickly specify what you did to verify proper functionality?

**Comment by jeffaco** [comment]
I'd like to see one more iteration with these fixes before commit, please.

Also, you probably want to swash your multiple commits into one for this?

File: gitinstall-hp.md (hunk 3 above)
Comment (niroyb) [inline gitinstall-hp.md:53]: No, I thought about uncompressing the files to have one less install step but it turns out uncompressing is fast and a 50% smaller footprint.

File: gitinstall-hp.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+## Installing via Depot packages
+
+This is the recommended install method if you have read access to the wsscfs network share.
+
+* Login as root on the target HPv3 system : `ssh root@myHPv3system`
+* Create a temporary folder to hold the depot files : `mkdir -p /root/GIT`
+* Copy over the depot packages to `/root/GIT`
+  * HPv3 ia64 : `\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3ia64`
+  * HPv3 parisc : `\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3risc`
```
Comment (jeffaco) [inline gitinstall-hp.md]: The above two links are bad, must start with two \ characters, like \\

File: gitinstall-hp.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+* [Installing via Internet](#installing-via-internet)
+  * [Caching Depot Packages](#caching-depot-packages)
+* [Installing via Source Code](#installing-via-source-code)
+* [Basic git functionality test](#basic-git-functionality-test)
+
+## Installing via Depot packages
+
+This is the recommended install method if you have read access to the wsscfs network share.
+
+* Login as root on the target HPv3 system : `ssh root@myHPv3system`
```
Comment (jeffaco) [inline gitinstall-hp.md:15]: You can probably assume the person knows how to log in as root! :smile: Okay to leave this, just chuckled a bit.

File: gitinstall-hp.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+* Create a temporary folder to hold the depot files : `mkdir -p /root/GIT`
+* Copy over the depot packages to `/root/GIT`
+  * HPv3 ia64 : `\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3ia64`
+  * HPv3 parisc : `\redmond\wsscfs\OSTCData\NixArchive\SourceCodeManagement\git\HP-UX\v3risc`
+* Unzip the compressed depot files : `gunzip /root/GIT/*.gz`
+* Install the git dependencies then git itself. The dependencies were [obtained using depothelper](#caching-depot-packages).
+
+For HPv3 ia64:
```
Comment (jeffaco) [inline gitinstall-hp.md]: So if this is for ia64, what about parisc? Do you have instructions for those?

Was there some way this list was generated? If so, you should comment on how to generate this list.

Lastly, for Solaris, I generated command scripts that you can run. Should you do that here too, or just leave it as is?

File: gitinstall-hp.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+* Test the installation by trying to clone a repo, create a commit in a new branch and push it upstream.
+* Cleanup : `rm -rf /root/GIT`
+
+## Installing via internet
+
+This method should be used if you don't already have access to the downloaded packages or if you want the latest versions of the packages installed. 
+* Get the correct depothelper version for your flavor of HP from [hpux.connect.org](http://hpux.connect.org.uk/hppd/hpux/Sysadmin/depothelper-2.00/)
+* Install depothelper for example: `swinstall -s depothelper-2.00-hppa-11.31.depot depothelper`
+* Install git with depothelper : `/usr/local/bin/depothelper git`
+* Test the installation by trying to clone a repo, create a commit in a new branch and push it upstream.
```
Comment (jeffaco) [inline gitinstall-hp.md]: Replace this line with something like:

[Test the installation](#basic-git-functionality-test)

File: gitinstall-hp.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+It is possible that the git binary installed through depothelper does not work because the configuration of your system is different. Simply rebuild git from source.
+* Get the appropriate HP ported version of git source code from [hpux.connect.org](http://hpux.connect.org.uk/hppd/hpux/Development/Tools/git-2.7.0/). It is the one that has the package type : Source Code.
+* Although the file is named "tar.gz" it might just be a regular tar file. Verify with `file git-2.7.0-src-11.31.tar.gz`
+* Untar the git source `tar -x git-2.7.0-src-11.31.tar.gz`
+* Configure git for your system : `cd git-2.7.0/; ./configure`
+* Build git : `make -j`
+* Make sure an old version of git is not present : `sudo swremove git`
+* Install git : `sudo make install`
+* Test the installation by trying to clone a repo, create a commit in a new branch and push it upstream.
```
Comment (jeffaco) [inline gitinstall-hp.md]: Replace this line with something like:

[Test the installation](#basic-git-functionality-test)

File: gitinstall-hp.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+## Basic git functionality test
+```shell
+git clone --recursive git@github.com:Microsoft/Build-SCXcore.git
+cd Build-SCXcore/
+git checkout -b test-branch
+echo test > test.txt
+git add test.txt
+git config --global user.email "you@example.com"
+git config --global user.name "Your Name"
+git commit -m "test message"
```
Comment (jeffaco) [inline gitinstall-hp.md:97]: Add `git push` to test pushing upstream, since you specifically called that out for testing.

**Comment by jeffaco** [comment]
Signed off, once you address my latest comments.

**Pull request closed without merging**